src/components/organisms/modals/EditableListModal/EditableListModal.tsx

```diff
@@ -0,0 +1,267 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { FC, useCallback, useEffect, useMemo, useState } from 'react'
+import Button, {
+  AppearanceTypes,
+  IconPositioningTypes,
+  SizeTypes,
+} from 'components/molecules/Button/Button'
+import ModalBase, {
+  ButtonPositionTypes,
+  ModalSizeTypes,
+  TitleFontTypes,
+} from 'components/organisms/ModalBase/ModalBase'
+import { useTranslation } from 'react-i18next'
+import DynamicForm, {
+  FieldProps,
+  InputTypes,
+} from 'components/organisms/DynamicForm/DynamicForm'
+import {
+  filter,
+  includes,
+  isEqual,
+  join,
+  map,
+  omitBy,
+  reduce,
+  size,
+  split,
+  toNumber,
+} from 'lodash'
+import { FieldPath, SubmitHandler, useForm } from 'react-hook-form'
+import { ReactComponent as Add } from 'assets/icons/add.svg'
+import { TagTypes } from 'types/tags'
+import useAuth from 'hooks/useAuth'
```

Comment:
**[non-blocker]** (For now)

You can deal with this in the departments task, where you plan to reuse this component, but currently you have the "TagTypes" type and Tag specific privilege checks inside the modal. Not sure if these will work with the department cases, so most likely you need to pass these as props as well.

But I haven't checked the spec for the department part, so not sure and either way it will be easier to deal with while dealing with departments